src/api_authentication_app_skeleton/src/actions/mixins/auth/api/action_helpers.cr

```diff
@@ -0,0 +1,23 @@
+require "jwt"
+
+module Auth::Api::ActionHelpers
+  def sign_in(user : Authentic::PasswordAuthenticatable)
+    context.response.headers.add "Authorization", user.generate_token
+  end
+
+  def token
+    context.request.headers["Authorization"]
+  end
+
+  def current_user?
+    user_from_token(token.not_nil!)
+  rescue KeyError | JWT::DecodeError | JWT::ExpiredSignatureError
```

Comment:
Rescuing KeyError can hide some nasty bugs. It's a fairly generic error so it could be raised if the Authorization header is missing, but it might also raise if any other key is missing. Instead I think something like this would be safer:

```crystal
def current_user?
  token.try do |value|
    user_from_token(value)
  end
rescue JWT::DecodeError | JWT::ExpiredSignatureError
  nil
end
```

Reply:
Yeah that makes sense, I'll take it out

Reply:
Okay I though I could use `.try` like this:
```
def current_user?
  token.try do |value|
    user_from_token(value)
  end
rescue JWT::DecodeError | JWT::ExpiredSignatureError
  nil
end
```
But it still throws `"ERROR: Missing hash key: HTTP::Headers::Key(@name=\"Authorization\") (KeyError)\n"` since token is trying to access the `Authorization` header:
```crystal
def token
  context.request.headers["Authorization"]
end
```

That's why I caught the `KeyError` before. What if I rescued the `KeyError` in the token method? like:

```crystal
def token
  context.request.headers["Authorization"]
  rescue KeyError
    nil
end
```
Since I'm only trying to access the `Authorization` key it shouldn't hide any other possible errors right?

Reply:
Ah I see. I think what you want is `...headers["Authorization"]?` (note the trailing `?`). That will tell the compiler it can return `nil` without raising. LMK if that does the trick

Reply:
Oh it looks like you did this. I didn't refresh so this still looked open :)